src/components/NotesMindMapView/NotesMindMapViewComponent.js

```diff
@@ -88,24 +92,45 @@ export class NotesMindMapViewComponent extends Component {
     let targetNote = noteStorage.findNoteById(targetNoteId);
 
     if (!targetNote) {
+      console.log("available");
+      console.log(this.props.changeVisNetworkNote);
       throw new Error('changeVisNetworkNote(): couldn\'t find targetNote in noteStorage by id: ', targetNoteId);
     }
 
     this.props.changeVisNetworkNote({ targetNote, visNetwork });
   };
 
   visNetworkClickHandler = event => {
-    this.closeNoteNameEditor();
-    if (VisNetworkHelper.clickedOnNote(event)) {
+    const { currentViewedNote } = this.state;
+    const { changeNoteParentMode } = this.props;
+
+    if(!changeNoteParentMode) 
+    {
+      this.closeNoteNameEditor();
+    }
+
+    if (VisNetworkHelper.clickedOnNote(event)) 
+    {
       let targetNoteId = VisNetworkHelper.getTargetNoteId(event);
-      this.noteClickHandler(targetNoteId);
+
+      if(changeNoteParentMode) 
+      {
+        this.props.changeParentNote({
+          note: currentViewedNote.id, 
+          parent: currentViewedNote.parent.id, 
+          target: targetNoteId, 
```

Comment:
1. You say `target`, but you pass `targetId`. Same for `parent` and `note`.
2. `parent`/`target` is a bit confusing. One may think that "parent" is the new parent to set: you call `changeParentNote` and pass `parent` as the desired parent for the note. Naming these props like `currentParent` and `newParent` would be easier to understand IMHO.
3. Can't we just pass the whole `note` instead of passing `note.id` and `note.parent.id` as two separate properties?